pkg/daemon/update.go

```diff
@@ -2200,6 +2409,18 @@ func (dn *Daemon) reboot(rationale string) error {
 	// and we wait for GracefulNodeShutdown
 	dn.rebootQueued = true
 	logSystem("reboot successful")
+	err := upgrademonitor.GenerateAndApplyMachineConfigNodes(
+		&upgrademonitor.Condition{State: mcfgalphav1.MachineConfigNodeUpdatePostActionComplete, Reason: string(mcfgalphav1.MachineConfigNodeUpdateRebooted), Message: "Node has rebooted"},
+		&upgrademonitor.Condition{State: mcfgalphav1.MachineConfigNodeUpdateRebooted, Reason: fmt.Sprintf("%s%s", string(mcfgalphav1.MachineConfigNodeUpdateRebooted), string(mcfgalphav1.MachineConfigNodeUpdateRebooted)), Message: "Upgrade required a reboot. Completed this as the post update action."},
```

Comment:
technically this isn't quite true since we're waiting for the node to shutdown which may or may not fail. Did we have a state before that stated a reboot is required already? Or is this the only instance